test/src/us/kbase/kbasefeaturevalues/test/KBaseFeatureValuesServerTest.java

```diff
@@ -195,61 +197,59 @@ public void testMainPipeline() throws Exception {
         String clustObj3Name = "clusters3";
         ExpressionMatrix data = new ExpressionMatrix().withType("log-ratio").withScale("1.0")
                 .withData(getSampleMatrix());
-        wscl.saveObjects(new SaveObjectsParams().withWorkspace(testWsName).withObjects(Arrays.asList(
+        getWsClient().saveObjects(new SaveObjectsParams().withWorkspace(testWsName).withObjects(Arrays.asList(
                 new ObjectSaveData().withName(exprObjName).withType("KBaseFeatureValues.ExpressionMatrix")
-                .withData(new UObject(data)))));
-        //Assert.assertNotNull("exprObjName ",testWsName + "/" + exprObjName);
-        //Assert.assertEquals("exprObjName ",testWsName + "/" + exprObjName);
-        //Assert.assertNotNull("estimObjName ", estimObjName);
-        //
+                        .withData(new UObject(data)))));
         /////////////// estimate K /////////////////
         EstimateKParams ekp = new EstimateKParams().withInputMatrix(testWsName + "/" +
                 exprObjName).withOutWorkspace(testWsName).withOutEstimateResult(estimObjName);
-        impl.estimateK(ekp, token, getContext());
-        ObjectData res1 = wscl.getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
-                .withName(estimObjName))).get(0);
-        EstimateKResult estKRes = res1.getData().asClassInstance(EstimateKResult.class);
+        EstimateKResult estKRes = impl.estimateK(ekp, token, getContext());
+
+        //ObjectData res1 = getWsClient().getObjects(Arrays.asList(new ObjectIdentity().withWorkspace(testWsName)
+        //        .withName(estimObjName))).get(0);
+        //EstimateKResult estKRes = res1.getData().asClassInstance(EstimateKResult.class);
         long k = estKRes.getBestK();
```

Comment:
Is there a reason the tests only check portions of the object?

Reply:
I also see there's many cases of this, I won't point them all out.

Reply:
for EstimateKResult there are only two relevant fields and those have always been tested

Reply:
Ok, it'd be good to document that. To a person who is a non-expert in the codebase it looks like parts of the data are ignored for no reason.

Reply:
wouldn't they just look in the object spec?
I added a comment line ...

is it standard practice to test null fields?
I added the only other possible test, for the empty additionalProperties

Reply:
> wouldn't they just look in the object spec?
I doubt it. I didn't. I'd think most people would just go 'why the heck is the object only partially tested?'

Even if they did look at the spec, a non-expert in this module would also probably not know which fields in the spec were important and which weren't.

> is it standard practice to test null fields?

I do *shrug*. If the field is supposed to be null and not some other value, that should be tested.

Reply:
> I added the only other possible test, for the empty additionalProperties

Oh, that I wouldn't bother about unless there's something you're expecting to be in there.

Reply:
based on your comments above I'm keeping the additionalProperties test ie "If the field is supposed to be null and not some other value, that should be tested."

Reply:
Probably not necessary IMO, but doesn't hurt for sure.